my-app/src/Components/GenerateNewText.js

```diff
@@ -0,0 +1,13 @@
+export async function GenerateNewText(textLength) {
```

Comment:
You should stick to the 3 Layer Architecture here as well.
Meaning:

Create a Service and a Repository.
TextService and TextRepository.

The Repository is responsible for communicating with the API.
The Service handles Business Logic (nothing for this case I guess).

Create the Service and Repository and pass them as props to the component needing them.